handlers/authentication.go

```diff
@@ -682,9 +683,16 @@ func AuthenticationInfo(w http.ResponseWriter, r *http.Request) {
 		response.LogoutEndpoint = metadata.LogoutEndpoint
 		response.LogoutRedirect = metadata.LogoutRedirect
 	case config.AuthStrategyOpenId:
-		response.AuthorizationEndpoint = fmt.Sprintf("https://dex.example.com:32000/auth?client_id=example-app&response_type=id_token&redirect_uri=%s&scope=%s&nonce=%s",
-			url.QueryEscape("http://172.17.0.2/kiali"),
-			"openid email profile",
+		scopes := strings.Join(conf.Auth.OpenId.Scopes, " ")
+		if !strings.Contains(scopes, "openid") {
+			scopes = "openid " + scopes
+		}
+
+		response.AuthorizationEndpoint = fmt.Sprintf("%s?client_id=%s&response_type=id_token&redirect_uri=%s&scope=%s&nonce=%s",
+			conf.Auth.OpenId.AuthorizationEndpoint,
+			url.QueryEscape(conf.Auth.OpenId.ClientId),
+			url.QueryEscape(httputil.GuessKialiURL(r)),
+			url.QueryEscape(scopes),
 			"asdf123456")
```

Comment:
hardcoded strings like this scare me now - what is this? If this is some kind of key or seed, then we need to randomize somewhere.

Reply:
Yes, `nonce` should be a random string. It's usage is to mitigate replay attacks. I'll replace this dummy string in a subsequent commit.